pkg/service/sync_task.go

```diff
@@ -208,6 +216,46 @@ func buildIdMap(sel SyncTask, recordSliceRef reflect.Value) map[string]struct{}
 		id := sel.ID(entryRef.Interface())
 		ids[id] = struct{}{}
 	}
-
 	return ids
 }
+
+func patchSelfTrade(ctx context.Context, db *sqlx.DB, sel SyncTask, recordSliceRef reflect.Value, ids *map[string]struct{}) error {
```

Comment:
map itself is already a pointer, so you can just pass it

Reply:
Done.